Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,424 @@
+    bool CreateGem::ValidateGemName()
+    {
+        bool gemSystemNameIsValid = true;
+        if (m_gemName->lineEdit()->text().isEmpty() || !m_gemName->lineEdit()->hasAcceptableInput())
+        {
+            gemSystemNameIsValid = false;
+        }
+        m_gemName->setErrorLabelVisible(!gemSystemNameIsValid);
+        return gemSystemNameIsValid;
+    }
+
+    bool CreateGem::ValidateFormNotEmpty(FormLineEditWidget* form)
+    {
+        bool formIsValid = !form->lineEdit()->text().isEmpty();
+        form->setErrorLabelVisible(!formIsValid);
+        return formIsValid;
+    }
+
+    bool CreateGem::ValidateRepositoryURL()
+    {
+        bool repositoryURLIsValid = true;
+        if (!m_repositoryURL->lineEdit()->text().isEmpty())
+        {
+            QUrl checkURL = QUrl(m_repositoryURL->lineEdit()->text(), QUrl::StrictMode);
+            repositoryURLIsValid = checkURL.isValid();
+            m_repositoryURL->setErrorLabelText(tr("Repository URL is not valid."));
+        }
+        m_repositoryURL->setErrorLabelVisible(!repositoryURLIsValid);
+        return repositoryURLIsValid;
+    }
+
+    void CreateGem::HandleBackButton()
+    {
+        if (m_tabWidget->currentIndex() > 0)
+        {
+            m_tabWidget->setCurrentIndex(m_tabWidget->currentIndex() - 1);
+        }
+
+        if (m_tabWidget->currentIndex() == 0)
+        {
+            m_backButton->setVisible(false);
+        }
+
+        m_nextButton->setText(tr("Next"));
+    }
+
+    void CreateGem::HandleNextButton()
+    {
+        if (m_tabWidget->currentIndex() == 0 && ValidateGemTemplateLocation())
+        {
+            m_backButton->setVisible(true);
+            m_tabWidget->setCurrentIndex(1);
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            bool gemNameValid = ValidateGemName();
+            bool gemDisplayNameValid = ValidateGemDisplayName();
+            bool licenseValid = ValidateFormNotEmpty(m_license);
+            if (gemNameValid && gemDisplayNameValid && licenseValid)
+            {
+                m_createGemInfo.m_displayName = m_gemDisplayName->lineEdit()->text();
+                m_createGemInfo.m_name = m_gemName->lineEdit()->text();
+                m_createGemInfo.m_summary = m_gemSummary->lineEdit()->text();
+                m_createGemInfo.m_requirement = m_requirements->lineEdit()->text();
+                m_createGemInfo.m_licenseText = m_license->lineEdit()->text();
+                m_createGemInfo.m_licenseLink = m_licenseURL->lineEdit()->text();
+                m_createGemInfo.m_documentationLink = m_documentationURL->lineEdit()->text();
+                m_createGemInfo.m_path = m_gemLocation->lineEdit()->text();
+                m_createGemInfo.m_features = m_userDefinedGemTags->lineEdit()->text().split(',');
+
+                m_tabWidget->setCurrentIndex(2);
+                m_nextButton->setText(tr("Create"));
+            }
+        }
+        else if (m_tabWidget->currentIndex() == 2)
```

Comment:
Can you create constants for these indices?
I am assuming 2 corresponds to the Create Gem button being selected?